blueprints/hr_assistant/date.py

```diff
@@ -0,0 +1,329 @@
+# -*- coding: utf-8 -*-
+"""This module contains the dialogue states for the 'date' domain in
+the MindMeld HR assistant blueprint application
+"""
+from .root import app
+from hr_assistant.general import (
+    _resolve_categorical_entities, _resolve_function_entity,
+    _resolve_extremes, _agg_function, _get_names, _not_an_employee
+)
+from dateutil.relativedelta import relativedelta
+import datetime
+import re
+
+
+@app.handle(intent='get_date')
+def get_date(request, responder):
+    """
+    If a user asks for a date related information of any person, this function returns
+    the required date. In case of a termination date related query, it also informs the user
+    of the reason for termination. For non-terminated employees, it informs the user about
+    the active current state of the employee in question.
+    """
+
+    # Look for name from the context of previous turn
+    name = request.frame.get('name')
+
+    # If user presents a new name, update the name in context
+    try:
+        name_ent = [e for e in request.entities if e['type'] == 'name']
```

Comment:
We should make a helper function called to just abstract such access patterns: `[e for e in request.entities if e['type'] == 'xyz']`

Reply:
`request.get_entities_by_type(type)`?

Reply:
lets do that